### PR TITLE
vm/vmss identity: use new patch API to handle identity update 

### PR DESCRIPTION
**Pull request by yugangw-msft**
This is to enable that users can add/remove an individual identity to a VM as long as they have permissions to that one. Before, they are required to have permissions to all identities assigned to the VM. 

@lmazuel , please let me know when you publish out the new azure-mgmt-compute and azure-mgmt-resource

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 1]
```diff
@@ -77,11 +77,10 @@ def default_api_version(self):
     'latest': {
         ResourceType.MGMT_STORAGE: '2017-10-01',
         ResourceType.MGMT_NETWORK: '2018-02-01',
-        ResourceType.MGMT_COMPUTE: SDKProfile('2017-12-01', {
+        ResourceType.MGMT_COMPUTE: SDKProfile('2018-06-01', {
             'resource_skus': '2017-09-01',
             'disks': '2018-04-01',
-            'snapshots': '2018-04-01',
-            'virtual_machine_run_commands': '2017-03-30'
```
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py]: Is this now in 2018-06-01?

File: src/command_modules/azure-cli-vm/HISTORY.rst [hunk 2]
```diff
@@ -2,10 +2,10 @@
...
 Release History
 ===============
-
-2.1.1
-+++++
-* Minor fixes.
+2.2.0
+++++++
+* BREAKING CHANGE: `vm/vmss create`, `vm/vmss identity assign/remove` will output `user assigned identities`
+                   in dictionary format
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/HISTORY.rst]: What did it output before?

Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/HISTORY.rst]: Before:
```bash
  "userAssignedIdentities": ["/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/e2e/providers/Microsoft.ManagedIdentity/userAssignedIdentities/e2e-id", "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/e2e/providers/Microsoft.ManagedIdentity/userAssignedIdentities/e2e-id2"]
```

After:
```bash
  "userAssignedIdentities": {
    "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/e2e/providers/Microsoft.ManagedIdentity/userAssignedIdentities/e2e-id": {
      "clientId": "59f15e9d-9aed-4121-b4d1-fe29fa13f1eb",
      "principalId": "24e5a32c-c88b-40f0-9868-11f846b8c280"
    },
    "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/e2e/providers/Microsoft.ManagedIdentity/userAssignedIdentities/e2e-id2": {
      "clientId": "0e8a38fa-c48e-4ab0-bbe4-7249cfd1c6d6",
      "principalId": "275b5b36-cec1-4d22-a097-ce40bc91ab5e"
    }
```
I originally wrote extra code to hide the new fields of 'clientId' and  'principalId', but backed out as those are new and important information users do want to see

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py (hunk 1 above)
Comment (yugangw-msft) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py]: yes

**Comment by troydai** [comment]
When are these SDK going to be released? Before that I assume this is do not merge?

**Comment by yugangw-msft** [comment]
@troydai, SDK will be released today, and correct, i will not merge this PR till that happens.

**Comment by lmazuel** [comment]
@yugangw-msft step 1 : https://pypi.org/project/azure-mgmt-resource/2.0.0/

**Comment by lmazuel** [comment]
@yugangw-msft step2 : https://pypi.org/project/azure-mgmt-compute/4.0.0/

**Pull request merged**